src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt

```diff
@@ -140,6 +142,25 @@ class LicenceService(
     return missingFields
   }
 
+  fun getAddressType(curfew: Curfew?, cas2Referral: Cas2Referral?, proposedAddress: ProposedAddress?, licenceId: Long?): AddressType? {
```

Comment:
mmh.. don't you know this already based on where you get the address from in `getAddress`? 

```kt
  val address = when {
      isCurfewApprovedPremisesRequired && !isCas2ApprovedPremisesRequired -> curfew?.approvedPremisesAddress
      isCas2ApprovedPremisesRequired -> cas2Referral?.approvedPremisesAddress
      isCas2Requested && isCas2Accepted -> cas2Referral?.bassOffer
      else -> proposedAddress?.curfewAddress
    }
```

first 3 are CAS and the last one is not CAS?

Reply:
thank you - addressed in the latest commit